### PR TITLE
chore(typing): fix tsdb method calls

### LINKED ISSUE #57604: Fix typing for `sentry.tasks.servicehooks`
Many files related to post-process are skipping type checks.

* Remove the file from [sentry/pyproject.toml](https://github.com/getsentry/sentry/blob/b7810a9dca4b57afd2858903a6a9ec7ab50cdead/pyproject.toml)
* Run `mypy sentry.tasks.servicehooks` and fix the typing errors

From https://github.com/getsentry/sentry/issues/55193

### PR DESCRIPTION
**Pull request by isabellaenriquez**
This PR fixes calls of tsdb methods so that `tsdb.backend` is used. 

Note: there are still some calls without the `backend` part, however, these were intentionally left out as there were other typing errors in those files. Thus, those calls can be fixed in separate PRs fixing the typing for those entire files.

Closes #57604

### CODE REVIEW

File: src/sentry/tasks/servicehooks.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def process_service_hook(servicehook_id, event, **kwargs):
 
     from sentry import tsdb
 
-    tsdb.incr(TSDBModel.servicehook_fired, servicehook.id)
+    tsdb.incr(TSDBModel.servicehook_fired, servicehook.id)  # type: ignore
```
Comment (asottile-sentry) [inline src/sentry/tasks/servicehooks.py:49]: this is not the correct fix -- this should use `tsdb.backend.incr`

**Review by asottile-sentry — CHANGES_REQUESTED** [review]

File: src/sentry/tasks/servicehooks.py (hunk 1 above)
Comment (asottile-sentry) [reply src/sentry/tasks/servicehooks.py:49]: please fix the other file too

Comment (isabellaenriquez) [reply src/sentry/tasks/servicehooks.py:49]: thanks for letting me know! will fix in both files

Comment (isabellaenriquez) [reply src/sentry/tasks/servicehooks.py:49]: looks like there's a few other files where the same was done (with a typing ignore comment or in a file that is skipped for typing) so i'll look into fixing those as well

**Review by asottile-sentry — APPROVED** [review]

**Review by snigdhas — APPROVED** [review]

**Pull request merged**